react/features/user-customization/actions.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+
+import { getLogger } from 'jitsi-meet-logger';
+
+import { doGetJSON } from '../base/util';
+
+import { SET_USER_CUSTOM_DATA } from './actionTypes';
+import { extractFqnFromPath } from './functions';
+
+const logger = getLogger(__filename);
+
+/**
+ * Fetches custom branding data.
+ * If there is no data or the request fails, sets the `customizationReady` flag
+ * so the defaults can be displayed.
+ *
+ * @returns {Function}
+ */
+export function fetchCustomBrandingData() {
+    return async function(dispatch: Function, getState: Function) {
+        const state = getState();
+        const baseUrl = state['features/base/config']._brandingDataUrl;
+        const { customizationReady } = state['features/user-customization'];
+
+        if (!customizationReady) {
+            const fqn = extractFqnFromPath(state['features/base/connection'].locationURL.pathname);
+
+            if (baseUrl && fqn) {
+                try {
+                    const res = await doGetJSON(`${baseUrl}?conferenceFqn=${encodeURIComponent(fqn)}`);
+
+                    return dispatch(setUserCustomData(res));
+                } catch (err) {
+                    logger.error('Error fetching branding data', err);
+                }
+            }
+
+            dispatch(setUserCustomData({
```

Comment:
why do we say it's ready even when no customization was fetched?

Reply:
We use this flag to mark the moment when it's safe to display the logo. So if the meeting does not have a custom logo or if the request fails we display the default data. 

This is done intentional so if there is no custom data we don't display the logo/background twice.